components/compositing/touch.rs

```diff
@@ -25,6 +25,8 @@ const FLING_MAX_SCREEN_PX: f32 = 4000.0;
 pub struct TouchHandler {
     pub state: TouchState,
     pub active_touch_points: Vec<TouchPoint>,
+    pub prevent_move: bool,
+    pub prevent_click: bool,
```

Comment:
I think it might be better to move this state into the TouchState, since it removes some error potential of forgetting to reset the values.

Reply:
I had an offline discussion with kongbai, and to summarize:

- We can't really move the prevent_move / click variables into the TouchState, since the event handler is async, and the response for eg. `touch_down` may only arrive after e.g. `touch_up`.
- This also means that this PR as is introduces the problem that `prevent_move` and `prevent_click` are susceptible to races, and can be overwritten by a followup touch event (e.g. 2nd touch_down arrives before previous touch_up event handler has finished) 
- We need to introduce something like a touch_event_id, so that we can preserve `prevent_default` results in a map, until the event_handler has finished processing the last event for a given id (events between a touch_down and a touch_up / cancel)